packages/flutter/test/material/text_button_test.dart

```diff
@@ -0,0 +1,996 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+
+import '../rendering/mock_canvas.dart';
+import '../widgets/semantics_tester.dart';
+
+void main() {
```

Comment:
Should there be a test for all 3 buttons that confirms the behavior of:

```
foo.style ?? theme.style ?? defaultStyle
```

Reply:
I've added basic tests for all three new themes.